sandy/sections/mf6.py

```diff
@@ -0,0 +1,515 @@
+    tape : `sandy.Endf6`
+        endf6 object containing requested section.
+    mat : `int`
+        MAT number.
+    mt : `int`
+        MT number.
+
+    Returns
+    -------
+    out : `dict`
+        Content of the ENDF-6 tape structured as nested `dict`.
+
+    Examples
+    --------
+    Since the outputs are very large, I am only going to check the first energy
+    of each of the law-dependent structures:
+    Law 1:
+    >>> tape = sandy.get_endf6_file("endfb_71", 'xs', 70140)
+    >>> test = read_mf6(tape, 725, 5)
+    >>> test["NK"][10010]["EGROUPS"][1e-05]
+     {'ND': 0,
+     'NA': 1,
+     'NW': 6,
+     'NEP': 2,
+     'Ep': [0.0, 1e-05],
+     'b': [100000.0, 0.0, 0.0, 0.0]}
+
+    Law 2:
+    >>> import pprint
+    >>> tape = sandy.get_endf6_file("endfb_71", 'xs', 10010)
+    >>> test = read_mf6(tape, 125, 102)
+    >>> test["NK"][10020]['AWP'] = round (test["NK"][10020]['AWP'], 5)
+    >>> pprint.pprint(test["NK"][10020])
+    {'AWP': 1.99626,
+     'E': array([1.e-05, 2.e+07]),
+     'LAW': 4,
+     'NP': [2],
+     'NR': [2],
+     'Y': array([1., 1.])}
+
+    Law 6:
+    >>> tape = sandy.get_endf6_file("endfb_71", 'xs', 10020)
+    >>> test = read_mf6(tape, 128, 16)
+    >>> test["NK"][10]['APSX'] = round (test["NK"][10]['APSX'],5)
+    >>> test["NK"][10]
+    {'AWP': 1.0,
+     'LAW': 6,
+     'NR': [2],
+     'NP': [2],
+     'E': array([3.339002e+06, 1.500000e+08]),
+     'Y': array([2., 2.]),
+     'APSX': 2.99862,
+     'NPSX': 3}
+
+    Law 7:
+    >>> tape = sandy.get_endf6_file("endfb_71", 'xs', 40090)
+    >>> test = read_mf6(tape, 425, 16)
+    >>> test["NK"][10]["EGROUPS"][1748830.0]['COSGROUPS'][-1.0]
+    {'NRP': [15],
+     'NEP': [2],
+     'EP_INT': array([  1092.99,   1093.  ,   3278.9 ,   7650.8 ,  12023.  ,  20766.  ,
+             29510.  ,  55741.  ,  71043.  ,  81973.  ,  90716.  ,  95088.  ,
+             99460.  , 101650.  , 101651.  ]),
+     'E_p': array([0.00000e+00, 1.16614e-06, 1.58588e-06, 1.54624e-06, 7.09710e-07,
+            1.79581e-07, 2.86776e-08, 0.00000e+00]),
+     'E_distr': array([7.40674e-07, 1.46654e-06, 1.61094e-06, 1.07195e-06, 4.02172e-07,
+            9.52648e-08, 4.69275e-09])}
+    """
+    mf = 6
+    df = tape._get_section_df(mat, mf, mt)
+    out = {"MAT": mat, "MF": mf, "MT": mt}
+    i = 0
+    C, i = sandy.read_cont(df, i)
+    out.update({
+                "ZA": C.C1,
+                "AWR": C.C2,
+                "LCT": C.L2,
+               })
+    subsections = {}
+    # To repeat the proces the number of subsection that we have (NK)
+    for a in range(C.N1):
+        # To repeat TAB1 structure for each NK subsection
+        T, i = sandy.read_tab1(df, i)
+        LAW = T.L2
+        ZAP = T.C1
+        LIP = T.L1
+        ZAM = zam.za2zam(ZAP, meta=LIP)
+        add = {
+                "AWP": T.C2,
+                "LAW": T.L2,
+                "NR": T.NBT,
+                "NP": T.INT,
+                "E": T.x,
+                "Y": T.y,
+                }
```

Comment:
MF6 is complicated and requires some explanation, can you add comments after each **relevant** entry to say what they are?
Example:
```
                "E": T.x,  # energy of ...
                "Y": T.y, # what are these values?
                }
```
Look at https://github.com/luca-fiorito-11/sandy/blob/develop/sandy/sections/mf8.py for reference

Reply:
I did, but i have problems with maximun characters in some lines.